src/history/hs_rec.c

```diff
@@ -910,10 +910,9 @@ __hs_delete_reinsert_from_pos(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, ui
     /*
      * If we find a key with a timestamp larger than or equal to the specified timestamp then the
      * specified timestamp must be mixed mode.
-     *
-     * FIXME-WT-9846: Change this back to WT_ASSERT_ALWAYS once WT-9846 is resolved
      */
-    WT_ASSERT(session, ts == 1 || ts == WT_TS_NONE);
+    WT_ASSERT_ALWAYS(
+      session, ts == 1 || ts == WT_TS_NONE, "out-of-order timestamp update detected");
 
```

Comment:
Will this check for column store fail? Should we exclude that?

Reply:
Yep good point. I'll limit this to row-store and add a FIXME

Reply:
So we don't have access to the page here to determine row/column, and we'd need to pass that info down at least 2 levels in the stack. What do you think about leaving this as a diagnostic assert and changing the FIXME to WT-10017?

Reply:
I think it is ok to change it as a diagnostic assert and pass the value to it.

Reply:
After a quick discussion we'll leave the assert as diagnostic and update the FIXME for WT-10017 as WT-10017 is expected to come in quickly, hitting the conditions for WT-9846 is occurring quite rarely, and we'd still like to catch OOO timestamps in the row store case.